cli/tools/test.rs

```diff
@@ -450,31 +479,63 @@ impl TestReporter for PrettyTestReporter {
   }
 
   fn report_summary(&mut self, summary: &TestSummary, elapsed: &Duration) {
-    if !summary.failures.is_empty() {
+    if !summary.failures.is_empty() || !summary.uncaught_errors.is_empty() {
+      #[allow(clippy::type_complexity)] // Type alias doesn't look better here
+      let mut failures_by_origin: BTreeMap<
+        String,
+        (Vec<(&TestDescription, &JsError)>, Option<&JsError>),
+      > = BTreeMap::default();
       let mut failure_titles = vec![];
-      println!("\nfailures:\n");
       for (description, js_error) in &summary.failures {
-        let failure_title = format!(
-          "{} {}",
-          &description.name,
-          colors::gray(format!(
-            "=> {}:{}:{}",
-            self
-              .to_relative_path_or_remote_url(&description.location.file_name),
-            description.location.line_number,
-            description.location.column_number
-          ))
-        );
-        println!("{}", &failure_title);
-        println!(
-          "{}: {}",
-          colors::red_bold("error"),
-          format_test_error(js_error)
-        );
-        println!();
-        failure_titles.push(failure_title);
+        let (failures, _) = failures_by_origin
+          .entry(description.origin.clone())
+          .or_default();
+        failures.push((description, js_error.as_ref()));
+      }
+      for (origin, js_error) in &summary.uncaught_errors {
+        let (_, uncaught_error) =
+          failures_by_origin.entry(origin.clone()).or_default();
+        let _ = uncaught_error.insert(js_error.as_ref());
+      }
+      println!("\nfailures:\n");
+      for (origin, (failures, uncaught_error)) in failures_by_origin {
+        for (description, js_error) in failures {
+          let failure_title = format!(
+            "{} {}",
+            &description.name,
+            colors::gray(format!(
+              "=> {}:{}:{}",
+              self.to_relative_path_or_remote_url(
+                &description.location.file_name
+              ),
+              description.location.line_number,
+              description.location.column_number
+            ))
+          );
+          println!("{}", &failure_title);
+          println!(
+            "{}: {}",
+            colors::red_bold("error"),
+            format_test_error(js_error)
+          );
+          println!();
+          failure_titles.push(failure_title);
+        }
+        if let Some(js_error) = uncaught_error {
+          let failure_title =
+            format!("[{}]", self.to_relative_path_or_remote_url(&origin));
```

Comment:
I suggest to remove these brackets and only use gray color so it looks similar to regular errors reports

Reply:
These are the titles for errors that most urgently need to be addressed, it's not appropriate to grey them. I considered even making them bold. How about this, I think it looks good:
```suggestion
            format!("{} (uncaught error)", self.to_relative_path_or_remote_url(&origin));
```

Reply:
Sure, I think that's better